xdem/spatialstats.py

```diff
@@ -820,213 +1028,384 @@ def vgm_sum(h, *args):
+        if model_name in valid_models:
+            exact_integral = exact_integrals[valid_models.index(model_name)]
+            squared_se += exact_integral(r, p, l)
+
+    # We sum all partial sill to get the total sill
+    total_sill = np.nansum(params_vgm.psill)
+    # The number of effective sample is the fraction of total sill by squared standard error
+    neff = total_sill/squared_se
+
+    return neff
+
+def _integrate_fun(fun: Callable, low_b: float, upp_b: float) -> float:
+    """
+    Numerically integrate function between an upper and lower bounds
+    :param fun: Function to integrate
+    :param low_b: Lower bound
+    :param upp_b: Upper bound
+
+    :return: Integral between lower and upper bound
+    """
+    return integrate.quad(fun, low_b, upp_b)[0]
+
+def neff_circular_approx_numerical(area: float, params_vgm: pd.DataFrame) -> float:
     """
-    Number of effective samples derived from numerical integration for any sum of variogram models a circular area
-    (generalization of Rolstad et al. (2009): http://dx.doi.org/10.3189/002214309789470950)
-    The number of effective samples N_eff serves to convert between standard deviation/partial sills and standard error
+    Number of effective samples derived from numerical integration for any sum of variogram models a circular area.
+    This is a generalization of Rolstad et al. (2009): http://dx.doi.org/10.3189/002214309789470950, which is verified
+    against exact integration of `neff_circular_approx_theoretical`. This approximation performs best for areas with
+    a shape close to that of a disk.
+    The input variogram parameters match the format of the dataframe returned by `fit_sum_variogram_models`, also
+    detailed in the parameter description to be passed manually.
+
+    The number of effective samples N_eff serves to convert between standard deviation and standard error
     over the area: SE = SD / sqrt(N_eff) if SE is the standard error, SD the standard deviation.
 
-    :param area: area
-    :param list_vgm: variogram functions to sum (range, model name, partial sill)
+    :param area: Area (in square unit of the variogram ranges)
+    :param params_vgm: Dataframe of variogram models to sum with three to four columns, "model" for the model types
+        (e.g., ["spherical", "matern"]), "range" for the correlation ranges (e.g., [2, 100]), "psill" for the partial
+        sills (e.g., [0.8, 0.2]) and "smooth" for the smoothness paramter if exists for this model (e.g., [None, 0.2]).
 
-    :returns: number of effective samples
+    :returns: Number of effective samples
     """
-    psill_tot = 0
-    for vario in list_vgm:
-        psill_tot += vario[2]
 
+    # Check input dataframe
+    _check_validity_params_vgm(params_vgm)
+
+    # Get the total sill from the sum of partial sills
+    total_sill = np.nansum(params_vgm.psill)
+
+    # Define the covariance sum function times the spatial lag, for later integration
     def hcov_sum(h):
```

Comment:
it seems like you have a similar function at L 972. Would it not be possible to use the same for both?